From 4424bb8cc334c9ad51a4e8dd4ddc9318e7b2517a Mon Sep 17 00:00:00 2001
From: "Satyanarayana, Sandhya" <sandhya.satyanarayana@ti.com>
Date: Fri, 29 Jun 2012 12:18:09 +0530
Subject: [PATCH 1042/1092] ARM: OMAP: AM33XX: PM: Fix bug related to ddr
 self-refresh disable

DDR which had been put in Self Refresh mode during suspend
has to be taken out of Self Refresh during resume.

The self-refresh mode was getting disabled, but due to the
incorrect shift not all the required bits were getting
cleared. Since the bit not cleared was never set this went
unnoticed till now.

This patch fixes the above issue and also updates the
relevant shadow register.

Signed-off-by: Satyanarayana, Sandhya <sandhya.satyanarayana@ti.com>
---
 arch/arm/mach-omap2/sleep33xx.S | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/sleep33xx.S b/arch/arm/mach-omap2/sleep33xx.S
index ed2156c..527fd66 100644
--- a/arch/arm/mach-omap2/sleep33xx.S
+++ b/arch/arm/mach-omap2/sleep33xx.S
@@ -459,8 +459,9 @@ emif_self_refresh_dis:
 	ldr	r0, emif_addr_virt
 	add	r0, r0, #EMIF4_0_SDRAM_MGMT_CTRL
 	ldr	r1, [r0]
-	bic	r1, r1, #(0x7 << 7)
+	bic	r1, r1, #(0x7 << 8)
 	str	r1, [r0]
+	str	r1, [r0, #4]
 
 	mov	r0, #7
 	ldmfd	sp!, {r4 - r11, pc}	@ restore regs and return
-- 
1.7.11.2

